src/libraries/System.Private.CoreLib/src/System/Globalization/CultureData.Windows.cs

```diff
@@ -714,5 +715,28 @@ internal bool IsReplacementCulture
                 return false;
             }
         }
+
+        internal static CultureData GetCurrentRegionData()
+        {
+            Span<char> geoIso2Letters = stackalloc char[10];
+
+            int geoId = Interop.Kernel32.GetUserGeoID(Interop.Kernel32.GEOCLASS_NATION);
+            if (geoId != Interop.Kernel32.GEOID_NOT_AVAILABLE)
+            {
+                int geoIsoIdLength = Interop.Kernel32.GetGeoInfo(geoId, Interop.Kernel32.GEO_ISO2, ref MemoryMarshal.GetReference(geoIso2Letters), geoIso2Letters.Length, Interop.Kernel32.EN_LANG_ID);
+                if (geoIsoIdLength != 0)
+                {
+                    geoIsoIdLength -= geoIso2Letters[geoIsoIdLength - 1] == 0 ? 1 : 0; // handle null termination and exclude it.
```

Comment:
So sometimes it's null terminated and sometimes not? Why does it not return ERROR_INSUFFICIENT_BUFFER when it has no room to null terminate?

Reply:
The doc is not ensuring we are returning null that is why I have to check to know if we got the null or not. 
We are requesting the 2-letters country code and I am providing a buffer of length 10 which is far enough to ensure not running into the ERROR_INSUFFICIENT_BUFFER case even with null.

Reply:
From what I can tell, this is always returning zero terminated string.

Would it be best to create the string from the pointer and not bother with using the returned size? I would be fine with no using the Span in that case because of there is no buffer math that can go wrong.

```
char* pGeoIso2Letters = stackalloc char[10];

if (Interop.Kernel32.GetGeoInfo(..., pGeoIso2Letters, ...) != 0)
{
    CultureData? cd = GetCultureDataForRegion(new string(pGeoIso2Letters), ...
```

Reply:
I would be more conservative checking null as the doc didn't ensure that and I am not sure what happen if we run on a different version of Windows.